packages/strapi-design-system/src/Grid/Grid.tsx

```diff
@@ -1,19 +1,23 @@
 import * as React from 'react';
 
-import PropTypes from 'prop-types';
 import styled from 'styled-components';
 
 import { GridContext } from './GridContext';
-import { Box } from '../Box';
+import { Box, BoxProps } from '../Box';
 import handleResponsiveValues from '../helpers/handleResponsiveValues';
 
-const GridWrapper = styled(Box)`
+export interface GridProps extends BoxProps<HTMLDivElement> {
+  gridCols: number;
+  gap: number | number[];
+}
+
+const GridWrapper = styled(Box)<GridProps>`
```

Comment:
```suggestion
const GridWrapper = styled(Box)<Required<Pick<GridProps, 'gridCols' | 'gap'>>>`
```

Whilst it's not technically wrong (although making two params optional will probably cause an issue) I think it's clearer just to do `Required<Pick<GridProps, 'gridCols' | 'gap'>>` only because it's the props we're using at this level so it's stricter but it's typically easier to loosen a type imo. Wdyt?